src/lib/consensus/proof_of_stake.ml

```diff
@@ -795,8 +867,19 @@ module Epoch_data = struct
         , prev_protocol_state_hash )
       else (curr_data.seed, curr_data.lock_checkpoint)
     in
+    let%map delegator = delegator_account curr_data in
+    let new_participated_stake =
+      if delegator.participated then curr_data.participated_stake
+      else
+        Option.value_exn
+          (Amount.add curr_data.participated_stake
```

Comment:
This is likely what we need to do here (I don't see an immediate alternative), but this is a little scary, as it seems realistic that this amount could overflow. It might be fine for now, as when we are properly tracking the participation bit, this amount will be capped to the total amount of currency in the system, which is fine.

Reply:
I made it an `Or_error`